### PR TITLE
Add exchange_buckets.

### PR DESCRIPTION
**Pull request by cmeiklejohn**
Add the ability to get the full bucket list, which will be used to front-load the transfer of buckets on latency-heavy links, reducing the number of required round trips.

### CODE REVIEW

File: src/riak_kv_index_hashtree.erl [hunk 1]
```diff
@@ -291,6 +298,14 @@ handle_call({exchange_bucket, Id, Level, Bucket}, _From, State) ->
                end,
                State);
 
+handle_call({exchange_buckets, Id}, _From, State) ->
+    apply_tree(Id,
+               fun(Tree) ->
+                       Result = hashtree:get_buckets(3, Tree),
```
Comment (cmeiklejohn) [inline src/riak_kv_index_hashtree.erl]: Change this back to `get_buckets/1`.

**Comment by cmeiklejohn** [comment]
cc: @jtuple

**Comment by Vagabond** [comment]
What even calls this nrw code?

**Comment by cmeiklejohn** [comment]
repl, yo.

**Comment by cmeiklejohn** [comment]
Updated commit to use `get_buckets/1` to prefetch all buckets.

**Comment by Vagabond** [comment]
+1

**Comment by cmeiklejohn** [comment]
Closing; not merging.

**Pull request closed without merging**